topics/appendix/tools/intellij_platform_gradle_plugin/tools_intellij_platform_gradle_plugin_jetbrains_runtime.md

```diff
@@ -0,0 +1,126 @@
+
+To use this approach, no extra steps are required but declare the IntelliJ Platform, see: [](tools_intellij_platform_gradle_plugin_dependencies_extension.md#target-platforms).
+
+## Obtained with IntelliJ Platform from Maven
+
+If the IntelliJ Platform is resolved with the IntelliJ Maven Repository instead of JetBrains Download CDN, it has no JetBrains Runtime (JBR) bundled.
+In such a case, it is necessary to provide it in the suitable version with the version read from the <path>[intellijPlatform]/dependencies.txt</path> file.
+
+This can be easily achieved by using the `jetbrainsRuntime()` repository helper pointing to [JetBrains Runtime GitHub Releases](https://github.com/JetBrains/JetBrainsRuntime/releases/) and dependency helper with no arguments provided:
+
+```kotlin
+repositories {
+  mavenCentral()
+  intellijPlatform {
+    defaultRepositories()
+    jetbrainsRuntime()
+    // ...
+  }
+}
+
+dependencies {
+  intellijPlatform {
+    intellijIdeaCommunity("%ijPlatform%")
+
+    jetbrainsRuntime()
+
+    // ...
+  }
+}
+```
+
+## Declared explicitly
+
+It is possible to explicitly specify JetBrains Runtime version, variant, or exact build with:
+
+```kotlin
+repositories {
+  mavenCentral()
+  intellijPlatform {
+    defaultRepositories()
+    jetbrainsRuntime()
+    // ...
+  }
+}
+
+dependencies {
+  intellijPlatform {
+    intellijIdeaCommunity("%ijPlatform%")
+
+    jetbrainsRuntime(version, variant, architecture)
+    // or
+    jetbrainsRuntimeExplicit(explicitVersion)
+
+    // ...
+  }
+}
+```
+
+Provided `version`, `variant`, and `architecture` parameters along with the `explicitVersion` are used to resolve the JetBrains Runtime archives published on [GitHub Releases](https://github.com/JetBrains/JetBrainsRuntime/releases/) page.
+
+To correctly understand the pattern, refer to the archive names in format `jbr_[prefix]-[jdk]-[os]-[arch]-b[build].tar.gz`, such as `jbr_jcef-21.0.3-osx-aarch64-b446.1.tar.gz`, which can be split into the following parts:
+
+- `prefix` = `jcef`
+- `jdk` = `21.0.3`
+- `os` = `osx`
+- `arch` = `aarch64`
+- `build` = `446.1`
+
+Note that the `version` parameter is actually a combination of JDK and build numbers in format `[jdk]b[build]`.
+
+To resolve the `jbr_jcef-21.0.3-osx-aarch64-b446.1.tar.gz` archive, you can use:
+- `jetbrainsRuntime("21.0.3b446.1", "osx", "aarch64")`
+- `jetbrainsRuntime("21.0.3b446.1")` (`os` and `architecture` are by default resolved with the current environment)
+- `jetbrainsRuntimeExplicit("jbr_jcef-21.0.3-osx-aarch64-b446.1")`
+
+See the [JetBrains Runtime releases page](https://github.com/JetBrains/JetBrainsRuntime/releases) for the list of available releases.
+
+## Foojay Toolchains Plugin
+
+Since Gradle `8.4`, it is possible to specify JetBrains as a known JVM vendor and instruct Gradle to search for JetBrains Runtime within available toolchains (see [Toolchains for JVM projects](https://docs.gradle.org/current/userguide/toolchains.html)).
```

Comment:
How does that differ from above? What is the use case? Which one should users prefer?

Reply:
It's explained below:

> Please note that the latest available JetBrains Runtime release is always resolved, which may lead to unexpected behaviors.